app/screens/PrivacyLock.tsx

```diff
@@ -1,34 +1,68 @@
-import { useEffect } from 'react'
-import { AppState, BackHandler } from 'react-native'
+import { useCallback, useEffect, useRef } from 'react'
+import { AppState, AppStateStatus, BackHandler } from 'react-native'
 import { Logging } from '../api'
 import { AppLastActiveTimestamp } from '../api/app_last_active'
-import { useLocalAuthContext } from '../contexts/LocalAuthContext'
+import { useLocalAuthContext, LocalAuthContext } from '../contexts/LocalAuthContext'
 
 export function PrivacyLock (): JSX.Element | null {
   const localAuth = useLocalAuthContext()
 
-  useEffect(() => {
-    AppState.addEventListener('change', nextState => {
-      if (nextState === 'background' || nextState === 'inactive') {
-        AppLastActiveTimestamp.set()
-          .then()
+  /**
+   * a dynamic event listener handling
+   * array of max length 2
+   * [0] get auto de-registered and spliced from the array when [1] pushed in
+   * as handler itself is callback with dependency, unable to bind a single instance or any singleton pattern
+   */
+  const prevHandlerRef = useRef<Array<(s: AppStateStatus) => void>>([])
+
+  const appStateHandler = useCallback((nextState: AppStateStatus) => {
```

Comment:
Split out to 3 different dependency hooks

1. handler itself - depend on privacy enabled status only
2. add app state listener - when handler (1) changed
3. remove previous listener - when a new handler registered (2)

some manual testing will be needed as OSs can behave differently.
- cold start
- app go background then come back (dev build timeout is 3 sec, preview will be 1 minute though, extra difficulty to do test)
- enable privacy lock in this launch (minimize, come back, require authentication)
- disable privacy lock in this launch (minimize, come back, do not require authentication)

Reply:
more test case (timeout mechanics)
- authenticate success, minimize, comeback after 1 minute: re-auth required
- authenticate success, minimize, come back within 1 minute: no re-auth required
- authenticate fail, minimize, come back within 1 minute: auth required